view/statistics-flow-certificates.js

```diff
@@ -0,0 +1,331 @@
+		now      = new db.Date();
+		dateFrom = new db.Date(serverQuery.dateFrom);
+		dateTo = serverQuery.dateTo || new db.Date(now.getFullYear(), now.getMonth(), now.getDate());
+		dateTo = new db.Date(dateTo);
+		duration = new Duration(dateFrom, dateTo);
+		mode = location.query.get('mode').map(function (mode) {
+			return mode;
+		}).value;
+		timeUnit = calculateDurationByMode(duration, mode);
+		pagination.count.value = Math.ceil(timeUnit / itemsPerPage);
+		pagination.current.value = location.query.get('page').map(function (page) {
+			return Number(page) || 1;
+		});
+		if (pagination.count.value > 1) {
+			offset = { from: ((pagination.current.value - 1) * itemsPerPage) };
+			offset.to = offset.from;
+			if (timeUnit - offset.from < itemsPerPage) {
+				offset.to += timeUnit - offset.from;
+			} else {
+				offset.to += itemsPerPage;
+			}
+
+			currentDate = dateFrom;
+			while (daysCount <= offset.to) {
+				if (daysCount === offset.from) {
+					dateFrom = currentDate;
+				}
+				if (daysCount === offset.to) {
+					dateTo = currentDate;
+				}
+				daysCount++;
+				switch (mode) {
+				case 'weekly':
+					currentDate = new db.Date(currentDate.getFullYear(),
+							currentDate.getMonth(), currentDate.getDate() + 7);
+					break;
+				case 'monthly':
+					currentDate = new db.Date(currentDate.getFullYear(),
+							currentDate.getMonth() + 1);
+					break;
+				case 'yearly':
+					currentDate = new db.Date(currentDate.getFullYear() + 1);
+					break;
+				default:
+					currentDate = new db.Date(currentDate.getFullYear(),
+							currentDate.getMonth(), currentDate.getDate() + 1);
+					break;
+				}
+			}
+		}
+		if (dateFrom) {
+			serverQuery.dateFrom = dateFrom.toJSON();
+		}
+		if (dateTo) {
+			serverQuery.dateTo = dateTo.toJSON();
+		}
+
+		queryServer(serverQuery).done(function (responseData) {
+			data.value = filterData(responseData);
+		});
+	});
+
+	section({ class: 'section-primary users-table-filter-bar' },
+		form({ action: '/flow/', autoSubmit: true },
+			div({ class: 'users-table-filter-bar-status' },
+				select(
+					{ id: 'service-select', name: 'service' },
+					option({ value: 'all', selected: location.query.get('service').map(function (value) {
+						return !value || value === 'all' ? 'selected' : null;
+					}) }, _("All services")),
+					list(db.BusinessProcess.extensions, function (ServiceType) {
+						var serviceName = getServiceName(ServiceType);
+						return option({
+							value: serviceName,
+							selected: location.query.get('service').map(function (value) {
+								var selected = (serviceName ? (value === serviceName) : (value == null));
+								return selected ? 'selected' : null;
+							})
+						}, ServiceType.prototype.label);
+					})
+				)),
+			div({ class: 'users-table-filter-bar-status' },
+				select(
+					{ id: 'certificate-select', name: 'certificate' },
+					option({ value: '', selected: location.query.get('certificate').map(function (value) {
```

Comment:
Same here, some reusable component will be good (e.g. in inpector view we should have also _certificates_ filter, even though now there is _registrations_ filter)